### ISSUE TITLE
Resource Processor: No space left on device

### ISSUE THREAD

**Issue by oliver7598** [microsoft/AzureTRE#1335]
Error:
`failed to register layer: ApplyLayer exit status 1 stdout:  stderr: write /cnab/app/terraform/.terraform/providers/registry.terraform.io/hashicorp/random/3.1.0/linux_amd64/terraform-provider-random_v3.1.0_x5: no space left on device `

As part of developing and testing new user resource templates, multiple variations of the created images are stored on disk in the resource processor (vmss-rp-porter-<tre_id>).  Eventually, the disk will reach its maximum capacity and this will prevent new user resources from being deployed.

Although this could immediately affect dev work it could eventually affect mature/long running TREs.

A clean up operation could be created to clean up older, redundant images stored on the resource processor.

As a short term solution, connection to the vmss-rp-porter scale set instance and manually removing redundant images frees up the disk space needed to continue creating new user resources.

**Steps to reproduce**
1. Register and deploy multiple different versions of user resources
2. Eventually an error will be returned stating `No space left on device`.
NOTE:  I had a total of 63 images on the resource processor when the issue arose.

**Comment by tamirkamara** [comment]
@oliver7598 63 seems like a high number and personally I never came across this problem. Is the number high because of work on many versions of images or part of normal operations?

**Comment by oliver7598** [comment]
> @oliver7598 63 seems like a high number and personally I never came across this problem. Is the number high because of work on many versions of images or part of normal operations?

@tamirkamara Yes this number was due working on the templates for both Windows and Linux user resources and releasing new versions each time to test their creations.  

I understand it is unlikely a production TRE may reach this number at the moment, it is still a possibility and one which will increase the longer a TRE is running as not just user resource images are stored on the resource processor.  

My current environment only has Guacamole, a Windows user resource and a Linux user resource and is at 31% capacity. (~3GB of which is the RP itself and the base workspace, the remaining ~5GB is Guacamole and its user resource images - currently one version of each) of the available 30GB.  

As the number of available services increases and new versions of new and existing images are released it becomes more and more likely this will be something which will be hit in the future in more than just development TREs.

As a minimum this will need to be documented in the troubleshooting guide although I feel the need for a cleanup operation/increased storage will also be needed.

**Comment by tamirkamara** [comment]
Thanks for replying @oliver7598. What do you think about the approach implemented in the linked PR?

**Comment by marrobi** [comment]
@tamirkamara Thanks for looking at this. Agree a clean up job is needed.

What I don't understand here is why so much space has been taken up in such a short space of time. Docker should be reusing the layer cache, hence 63 images with only small code changes should not eat that much space.

**Comment by tamirkamara** [comment]
@marrobi looking at @oliver7598 PR, I think the reason is that he has many files in play and while developing/testing probably many small changes were made and eventually brought that high number of bundle images...

**Comment by oliver7598** [comment]
@tamirkamara I've no issue with this PR and it should hopefully prevent this from happening in non-dev environments 

@marrobi The 63 images were created when working on both Linux RDP and Linux/Windows shared storage, both of which were tested on 2 base images per OS (Ubuntu 18.04 and Ubuntu 18.04 DSVM / Windows 10, Windows 2019 Server DSVM)

They also included a switch form using the generic azurerm_virtual_machine to azurerm_linux_virtual_machine and azurerm_windows_virtual_machine if this could have also affected the increase in usage.
